File: 3018-make-string-a-subsequence-using-cyclic-increments/3018-make-string-a-subsequence-using-cyclic-increments.c
bool canMakeSubsequence(char* str1, char* str2) {
    int cnt=strlen(str2);
    int j=0;
    for(int i=0;str1[i]!='\0' && str2[j]!='\0';i++){
        if(str1[i]==str2[j]){
            j++;
        }else{
            char tem=str1[i];
            if(tem=='z') tem='a';
            else tem++;
            if(tem==str2[j]) j++;
        }
    }
    if(j>=cnt) return true;
    return false;
}

File: 3018-make-string-a-subsequence-using-cyclic-increments/3018-make-string-a-subsequence-using-cyclic-increments.js
/**
 * @param {string} str1
 * @param {string} str2
 * @return {boolean}
 */
var canMakeSubsequence = function(str1, str2) {
    let cnt=str2.length;
    let j=0;
    for(let i=0;i<str1.length && j<str2.length;i++){
        if(str1[i]===str2[j]) j++;
        else{
            let nextChar = str1[i]==='z'?'a':String.fromCharCode(str1.charCodeAt(i)+1);
            if(nextChar===str2[j]){
                j++;
            }
        }
    }
    return j>=cnt;
};